### PR TITLE
feat: add vultr container registry resource

### PR DESCRIPTION
**Pull request by im6h**
## Description
- Based on Vultr API, add resource container registry and data container registry
- Add docs and examples to use the new resource
### Checklist:
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you linted your code locally prior to submission?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by im6h** [comment]
I fixed golint and added resources to terrafrom code
@optik-aper Would you like to check it again? Thanks

**Comment by Stealthmate** [comment]
I'd really like to see this merged!
I'm in the process of writing TF manifests for all my stuff on Vultr, and container registry is one of the first things I need to have in order to bootstrap everything else.

I'm no Go expert, but I can hack on this and try to resolve the conflicts if needed.

**Comment by optik-aper** [comment]
Sorry for the delay!  I'll test this out and merge it this afternoon.

File: vultr/resource_vultr_container_registry_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccVultrContainerRegistry(t *testing.T) {
+	crName := acctest.RandomWithPrefix("tf-cr-cr")
```
Comment (optik-aper) [inline vultr/resource_vultr_container_registry_test.go:14]:
```
    resource_vultr_container_registry_test.go:17: Step 1/1 error: Error running apply: exit status 1

        Error: error creating container registry: {"error":"Registry Name can only be lowercase alpha numeric characters","status":422}

          with vultr_container_registry.foo,
          on terraform_plugin_test.tf line 2, in resource "vultr_container_registry" "foo":
           2:           resource "vultr_container_registry" "foo" {
```

Comment (optik-aper) [reply vultr/resource_vultr_container_registry_test.go:14]: Probably should add some validation for this in the provider, but that can happen later

File: vultr/resource_vultr_container_registry.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+				Required: true,
+			},
+			"public": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"urn": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
```
Comment (optik-aper) [inline vultr/resource_vultr_container_registry.go]: Is not `optional`

File: vultr/resource_vultr_container_registry.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+			},
+			"storage": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Optional: true,
```
Comment (optik-aper) [inline vultr/resource_vultr_container_registry.go]: Is not `optional`

File: vultr/resource_vultr_container_registry.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+			},
+			"root_user": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Optional: true,
```
Comment (optik-aper) [inline vultr/resource_vultr_container_registry.go]: Is not `optional`

File: vultr/resource_vultr_container_registry.go [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+			},
+			"date_created": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
```
Comment (optik-aper) [inline vultr/resource_vultr_container_registry.go]: Is not `optional`

**Review by optik-aper — CHANGES_REQUESTED** [review]
This works for the most part.  The tests aren't valid based on the requirements of the API, so that should be fixed.  Some of the data schema isn't completely accurate either.  Ideally this would include some documentation in the `website/docs/r` directory, but that can be added later.

A `data_source` for container registry would be nice to have, but not crucial to the basic functionality here.  Another nice thing to have would be validation on the name field to ensure that the limited set of characters isn't allowed through.  But neither of those are blockers for this PR.

File: vultr/resource_vultr_container_registry_test.go (hunk 1 above)
Comment (im6h) [reply vultr/resource_vultr_container_registry_test.go:14]: I followed the Vultr API documentation, perhaps I forgot to validate the field. Thanks for your feedback.
https://www.vultr.com/api/#tag/Container-Registry/operation/create-registry

**Comment by im6h** [comment]
> I'd really like to see this merged! I'm in the process of writing TF manifests for all my stuff on Vultr, and container registry is one of the first things I need to have in order to bootstrap everything else.
> 
> I'm no Go expert, but I can hack on this and try to resolve the conflicts if needed.

Sure. I think we can make PR better before merging into the main

File: vultr/resource_vultr_container_registry_test.go (hunk 1 above)
Comment (im6h) [reply vultr/resource_vultr_container_registry_test.go:14]: Maybe we can transform the name to lowercase before sending the create request.

**Comment by F21** [comment]
Is there anything I can do to help with this one? Pretty keen to see this merged.

**Comment by optik-aper** [comment]
> Is there anything I can do to help with this one? Pretty keen to see this merged.

I just tested things out and it's working.  There are a few things I'd like to see cleaned up, and the name validation currently won't account for names with non-alphanumeric characters, but I will address that in a separate PR.  I'll get a new release out today or tomorrow.

**Review by optik-aper — APPROVED** [review]
Approved with some updates to follow

**Pull request merged**